src/utils/dev.js

```diff
@@ -1,118 +1,130 @@
-// reusable code for netlify dev
-// bit of a hasty abstraction but recommended by oclif
 const process = require('process')
 
+const fromEntries = require('@ungap/from-entries')
 const chalk = require('chalk')
 
-const {
-  NETLIFYDEVLOG,
-  // NETLIFYDEVWARN,
-  NETLIFYDEVERR,
-} = require('./logo')
-/**
- * inject environment variables from netlify addons and buildbot
- * into your local dev process.env
- *
- * ```
- * // usage example
- * const { api, site } = this.netlify
- * if (site.id) {
- *   const addonUrls = await addEnvVariables(api, site)
- *   // addonUrls is only for startProxy in netlify dev:index
- * }
- * ```
- */
-const addEnvVariables = async function (api, site) {
-  /** from addons */
-  const addonUrls = {}
-  const addons = await api.listServiceInstancesForSite({ siteId: site.id }).catch((error) => {
-    console.error(error)
-    switch (error.status) {
-      default:
-        console.error(
-          `${NETLIFYDEVERR} Error retrieving addons data for site ${chalk.yellow(
-            site.id,
-          )}. Double-check your login status with 'netlify status' or contact support with details of your error.`,
-        )
-        process.exit()
-    }
-  })
-  if (Array.isArray(addons)) {
-    addons.forEach((addon) => {
-      addonUrls[addon.slug] = `${addon.config.site_url}/.netlify/${addon.slug}`
-      for (const key in addon.env) {
-        const msg = () =>
-          console.log(`${NETLIFYDEVLOG} Injected ${chalk.yellow.bold('addon')} env var: ${chalk.yellow(key)}`)
-        process.env[key] = assignLoudly(process.env[key], addon.env[key], msg)
-      }
-    })
+const { loadDotEnvFiles } = require('./dot-env')
+const { NETLIFYDEVLOG } = require('./logo')
+
+const ERROR_CALL_TO_ACTION =
+  "Double-check your login status with 'netlify status' or contact support with details of your error."
+
+const getSiteData = async ({ api, site, error }) => {
```

Comment:
At some point, we should rename this `error()` function passed to all commands. It it confusingly named and clashes with actual error instances. Using a verb like `fail()` might be better.